content/spire/try/oidc-federation-aws.md

```diff
@@ -0,0 +1,415 @@
+---
+title: AWS OIDC Authentication
+description: Using OIDC to Authenticate SPIRE to AWS S3 on Kubernetes
+weight: 4
+toc: true
+aliases:
+menu:
+  spire:
+    weight: 50
+    parent: 'spire-try'
+---
+
+This tutorial builds on the [Kubernetes Quickstart](/spire/try/getting-started-k8s/) guide to describe how a SPIRE identified workload can, using a JWT-SVID, authenticate to Amazon AWS APIs, assume an AWS IAM role, and retrieve data from an AWS S3 bucket. This avoids the need to create and deploy AWS IAM credentials with the workload itself.
+
+In this tutorial you will learn how to:
+
+* Deploy the OIDC Discovery Provider Service
+* Create the required DNS A record to point to the Discovery Provider document
+* Create a sample AWS identity provider, policy, role, and S3 bucket
+* Test access to the S3 bucket
+
+## Prerequisites
+
+Note the following required accounts, prerequisites, and limitations before starting this tutorial:
+
+* You'll need access to the Kubernetes environment that you configured when going through [Kubernetes Quickstart](/spire/try/getting-started-k8s/) 
+* You'll need access to the AWS console to configure an identity provider, policy, role, and S3 bucket
+* This tutorial cannot run on Minikube as the Kubernetes environment must be network accessible by AWS
+* You'll need the ability to configure a DNS A record for the SPIRE OIDC Discovery Provider document location (see next section)
+
+### Required DNS A Record for the OIDC Discovery Provider IP Address
+
+The SPIRE OIDC Discovery Provider provides a URL to the location of the discovery document specified by the OIDC protocol. After starting the spire-oidc service you'll need to configure a DNS A record to point to the external IP address of that service. In the YAML files set up for this tutorial, replace MY\_DISCOVERY\_DOMAIN with the FQDN of the planned DNS hostname. The FQDN value for MY\_DISCOVERY\_DOMAIN does not need to correspond to an existing domain on your network. It is specific to the Kubernetes environment.
+
+To get the external IP address required for the A record, you will need to proceed through the steps in this tutorial until you start the spire-oidc service and verify the external IP address in the section [Verify That spire-oidc has an External Service Address](#verify-that-spireoidc-has-an-external-service-address). The DNS A record should take the following form:
+
+```console
+MY_DISCOVERY_DOMAIN          A        EXTERNAL-IP for spire-oidc service
+```
+
+For example:
+```console
+oidc-discovery.example.org   A        93.184.216.34
+```
+
+{{< info >}}
+As with any change to DNS, it will take minutes or hours for the new A record to propagate to DNS servers. This tutorial will not work until the A record is propagated. Once the A record is propagated, you should be able to view the discovery document at https://MY_DISCOVERY_DOMAIN/.well-known/openid-configuration .
```

Comment:
I'm not sure this is true. Initial records are picked up quickly (since these are cache misses). But changes will require the TTL to expire, which by default is 3600 seconds (1 hour). This can be avoided by either (a) getting it right the first time, or (b) asking the user to specify a low TTL when creating the entry with their DNS provider.